instrumentation/hystrix-1.4/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/hystrix/HystrixTest.java

```diff
@@ -0,0 +1,181 @@
+import com.netflix.hystrix.HystrixCommandGroupKey;
+import com.netflix.hystrix.HystrixCommandProperties;
+import io.opentelemetry.instrumentation.testing.junit.AgentInstrumentationExtension;
+import io.opentelemetry.instrumentation.testing.junit.InstrumentationExtension;
+import io.opentelemetry.sdk.trace.data.StatusData;
+import java.util.Objects;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.function.Function;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+class HystrixTest {
+
+  @RegisterExtension
+  protected static final InstrumentationExtension testing = AgentInstrumentationExtension.create();
+
+  @ParameterizedTest
+  @MethodSource("provideCommandActionArguments")
+  void testCommands(Function<HystrixCommand<String>, String> operation) {
+
+    HystrixCommand<String> command =
+        new HystrixCommand<String>(setter()) {
+          @Override
+          protected String run() throws Exception {
+            return tracedMethod();
+          }
+
+          private String tracedMethod() {
+            testing.runWithSpan("tracedMethod", () -> {});
+            return "Hello!";
+          }
+        };
+
+    String result = testing.runWithSpan("parent", () -> operation.apply(command));
+    assertThat(Objects.equals(result, "Hello!")).isTrue();
+
+    testing.waitAndAssertTraces(
+        trace ->
+            trace.hasSpansSatisfyingExactly(
+                span -> span.hasName("parent").hasNoParent(),
+                span ->
+                    span.hasName("ExampleGroup.HystrixTest$1.execute")
+                        .hasParent(trace.getSpan(0))
+                        .hasAttributesSatisfyingExactly(
+                            equalTo(stringKey("hystrix.command"), "HystrixTest$1"),
+                            equalTo(stringKey("hystrix.group"), "ExampleGroup"),
+                            equalTo(booleanKey("hystrix.circuit_open"), false)),
+                span -> span.hasName("tracedMethod").hasParent(trace.getSpan(1))));
+  }
+
+  @ParameterizedTest
+  @MethodSource("provideCommandActionArguments")
+  void testCommandFallbacks(Function<HystrixCommand<String>, String> operation) {
+
+    HystrixCommand<String> command =
+        new HystrixCommand<String>(setter()) {
+          @Override
+          protected String run() throws Exception {
+            throw new IllegalArgumentException();
+          }
+
+          @Override
+          protected String getFallback() {
+            return "Fallback!";
+          }
+        };
+
+    String result = testing.runWithSpan("parent", () -> operation.apply(command));
+    assertThat(Objects.equals(result, "Fallback!")).isTrue();
```

Comment:
To me using `Object.equals` here doesn't make sense, try `assertThat(result).isEqualTo("Hello!");`